enhancements/insights/insights-runtime-extractor.md

```diff
@@ -0,0 +1,647 @@
+Note over V: Volume shared between the extractor<br>and the exporter containers
+EXP->>IRE: Open a new TCP connection
+IRE->>V: create a /data/out-${timestamp} dir to collect all fingerprints output
+IRE->>IRE: call /crictl ps to list all containers runnnin on the worker node<br>and get their pod name and namespaces
+loop For each running container
+    IRE->>IRE: Call /crictl inspect to get the root PID of the container
+    IRE->>IRE: Get process info from /proc (name, cmd-line, env)
+    IRE->>V: Store the process fingerprint
+    create participant C as child process
+    IRE->>C: Fork the extractor process
+    C->>C: Enter namespaces (mnt, pid) of the container process
+    C->>C: Switch user
+    Note over C: executing inside the namespaces<br>of the running container process
+    C->>C: Find all all fingerprints executables that<br> can be applied to the process
+    loop For each applied fingerprint
+       C->>V: Execute the fingerprint and store its output in the timestamp dir
+    end
+    destroy C
+    IRE-xC: Wait until the child process is finished...
+end
+IRE->>EXP: Return the path of the timestamp dir
+IRE-xEXP: Close the TCP connection
+```
+
+
+
+#### insights-runtime-extractor Constraints
+
+The insights-runtime-extractor `extractor` is sensitive software that requires high privileges to function.
+It needs to run as privileged and access the host table of Worker Nodes to read their `/proc` tables and run as root to be able to enter in process namespaces.
+
+A specific `SecurityContextConstraints` should be added to the `openshift-insights` to deploy the insights-runtime-extractor with the required permissions.
+
+In order to minimize any security risks, its code will be audited for security and it image will be as minimal as possible.
+
+The `extractor` image contains:
+
+* The TCP server to trigger runtime info extraction (written in Rust)
+  * Access to this server is constrained by liseting only to the loopback address. Only clients in the same pod will be able to connect to it.
+*  The `crictl` tool to find the running containers and their root PID
+* “Fingerprints” which are self-contained executables that run in the container’s process namespaces to identify what the container is running.
+
+
+
+
+The `exporter` component runs an unprivileged HTTP server. When it receives a `GET /gather-runtime-info`, it triggers an extraction by connecting to the `extractor` that replies
+with a path to a dir on the shared volume. The exporter then collects the raw data from that directory, bundle them in a JSON payload sent as the HTTP response.
+Connections to the `exporter` HTTP server will be fronted by `kube-rbac-proxy` so that only the Insights Operator is allowed to query it.
+
+The `exporter` image would contain:
+
+* The HTTP server (written in Go)
+
+Due to the self-contained nature of these executables, the 2 images that compose the insights-runtime-extractor are built from `scratch` (and not from an existing base image).
```

Comment:
May need to use a minimal ubi image for FIPS.

Reply:
Thanks, I've amended the RFE. I put the FIPS compliance requirement in the `GA` stage but if you think that this must done for `TechPreview` please let me know.